synapse/lib/sqlite.py

```diff
@@ -0,0 +1,24 @@
+import sqlite3
+
+import synapse.lib.db as s_db
+
+'''
+Integration utilities for sqlite db pools.
+'''
+
+# turn on db cache sharing
+sqlite3.enable_shared_cache(1)
+
+def pool(size, path, **kwargs):
+    '''
+    Create an sqlite connection pool.
+
+    Args:
+        size (int): Number of connections in the pool
+    '''
+    def ctor():
+        db = sqlite3.connect(path, check_same_thread=False)
+        db.cursor().execute('PRAGMA read_uncommitted=1').close()
```

Comment:
Are there any other options we would like to change in here? It seems that this file is unused, other than in unit tests.

Reply:
We'll be migrating the cortex storage layers to using this object because it allows more granular reader vs writer control for transactions.  The current use case is the synapse-geospace module ( which uses it heavily :D )